Cppcheck errors reported in app/bpcat.c

**Describe the bug**
Cppcheck reports the following issue in app/bpcat.c:

```
app/bpcat.c:648:86: error: Subtracting pointers that point to different objects [comparePointers]
            status = bplib_send(desc, &msg_buffer, &msg_buffer.content[data_fill_sz] - (uint8_t *)&msg_buffer,
                                                                                     ^
app/bpcat.c:75:14: note: Variable declared here.
    uint8_t  content[BPCAT_DATA_MESSAGE_MAX_SIZE];
             ^
app/bpcat.c:648:52: note: Address of variable taken here.
            status = bplib_send(desc, &msg_buffer, &msg_buffer.content[data_fill_sz] - (uint8_t *)&msg_buffer,
                                                   ^
app/bpcat.c:554:25: note: Variable declared here.
    bpcat_msg_content_t msg_buffer;
                        ^
app/bpcat.c:648:99: note: Address of variable taken here.
            status = bplib_send(desc, &msg_buffer, &msg_buffer.content[data_fill_sz] - (uint8_t *)&msg_buffer,
                                                                                                  ^
app/bpcat.c:648:86: note: Subtracting pointers that point to different objects
            status = bplib_send(desc, &msg_buffer, &msg_buffer.content[data_fill_sz] - (uint8_t *)&msg_buffer,
```

**To Reproduce**
Execute cppcheck

**Expected behavior**
Should pass cleanly

**System observed on:**
Ubuntu 22.04

**Additional context**
Both types should be `uint8_t*` however this code might be clearer by using `offsetof()` instead.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/app/bpcat.c
+++ b/app/bpcat.c
@@ -645,7 +645,7 @@ static void *app_in_entry(void *arg)
                 send_ready = true;
             }
 
-            status = bplib_send(desc, &msg_buffer, &msg_buffer.content[data_fill_sz] - (uint8_t *)&msg_buffer,
+            status = bplib_send(desc, &msg_buffer, offsetof(bpcat_msg_content_t, content[data_fill_sz]),
                                 BPCAT_MAX_WAIT_MSEC);
             if (status == BP_SUCCESS)
             {

